esmvalcore/preprocessor/_regrid_iris_esmf_regrid.py

```diff
@@ -0,0 +1,233 @@
+"""Iris-esmf-regrid based regridding scheme."""
+from __future__ import annotations
+
+from collections.abc import Iterable
+from typing import Any, Literal
+
+import dask
+import dask.array as da
+import iris.cube
+import iris.exceptions
+import numpy as np
+from esmf_regrid.schemes import (
+    ESMFAreaWeightedRegridder,
+    ESMFBilinearRegridder,
+    ESMFNearestRegridder,
+)
+
+METHODS = {
+    'conservative': ESMFAreaWeightedRegridder,
+    'bilinear': ESMFBilinearRegridder,
+    'nearest': ESMFNearestRegridder,
+}
+
+
+def _get_dims_along_axes(
+    cube: iris.cube.Cube,
+    axes: Iterable[Literal["T", "Z", "Y", "X"]],
+) -> tuple[int, ...]:
+    """Get a tuple with the dimensions along one or more axis."""
+
+    def _get_dims_along_axis(cube, axis):
+        try:
+            coord = cube.coord(axis=axis, dim_coords=True)
+        except iris.exceptions.CoordinateNotFoundError:
+            try:
+                coord = cube.coord(axis=axis)
+            except iris.exceptions.CoordinateNotFoundError:
+                return tuple()
+        return cube.coord_dims(coord)
+
+    dims = {d for axis in axes for d in _get_dims_along_axis(cube, axis)}
+    return tuple(sorted(dims))
```

Comment:
I think this code could also added to the `preprocessor._shared` module. It's very general and could be useful for many preprocessors, not just regridding.

Reply:
Done in 42123c0